qt/python/mantidqt/widgets/workspacedisplay/matrix/table_view_model.py

```diff
@@ -81,6 +86,14 @@ def __init__(self, ws, model_type):
         else:
             raise ValueError("Unknown model type {0}".format(self.type))
 
+    def get_max_column_count(self):
```

Comment:
Could we have some docs for the new function?

Also I wonder if we could try `ws.blocksize()` first and do the slow loop if that fails to keep it fast for the more common, non-ragged case?